fastapi_asyncapi/schema.py

```diff
@@ -0,0 +1,260 @@
+    bindings: Optional[Bindings]
+    traits: Optional[OperationTraits]
+    message: Optional[Message]
+
+
+Subscribe = Operation
+Publish = Operation
+
+
+class Parameter(BaseModel):
+    description: Optional[str]
+    schema: Optional[Schema]
+    location: Optional[str]
+
+
+Parameters = List[Dict[str, Union[Parameter, Reference]]]
+
+
+ChannelBindings = Dict[str, Any]
+
+
+class ChannelItem(BaseModel):
+    ref: Optional[str] = Field(alias="$ref")
+    description: Optional[str]
+    servers: Optional[List[str]]
+    subscribe: Optional[Subscribe]
+    publish: Optional[Publish]
+    parameters: Optional[Parameters]
+    bindings: Optional[Union[ChannelBindings, Reference]]
+
+
+Security = Dict[str, List[str]]
+Channels = Dict[str, ChannelItem]
+SecurityRequirement = Dict[str, List[str]]
+ServerBinding = Dict[str, Any]
+
+
+class Server(BaseModel):
+    url: str
+    protocol: Protocol
+    protocolVersion: Optional[str]
+    description: Optional[str]
+    variables: Optional[Dict[str, ServerVariable]]
+    security: Optional[SecurityRequirement]
+    bindings: Optional[Union[ServerBinding, Reference]]
+
+
+class OAUHTFlow(BaseModel):
+    authorizationUrl: str
+    tokenUrl: str
+    refreshUrl: Optional[str]
+    scopes: Dict[str, str]
+
+
+class OAUHTFlows(BaseModel):
+    implicit: Optional[OAUHTFlow]
+    password: Optional[OAUHTFlow]
+    clientCredentials: Optional[OAUHTFlow]
+    authorizationCode: Optional[OAUHTFlow]
+
+
+class SecurityScheme(BaseModel):
+    type: str
+    description: Optional[str]
+    name: str
+    _in: str = Field(alias="in")
+    scheme: str
+    bearerFormat: Optional[str]
+    flows: OAUHTFlows
+    openIdConnectUrl: str
+
+
+ChannelBinding = Dict[str, Any]
+
+
+class Components(BaseModel):
+    schemas: Optional[Dict[str, Union[Schema, Reference]]]
+    servers: Optional[Dict[str, Union[Server, Reference]]]
+    channels: Optional[Dict[str, ChannelItem]]
+    messages: Optional[Dict[str, Union[Message, Reference]]]
+    securitySchemes: Optional[Dict[str, Union[SecurityScheme, Reference]]]
+    parameters: Optional[Dict[str, Union[Parameter, Reference]]]
+    correlationIds: Optional[Dict[str, Union[CorrelationID, Reference]]]
+    operationTraits: Optional[Dict[str, Union[OperationTraits, Reference]]]
+    messageTraits: Optional[Dict[str, Union[MessageTrait, Reference]]]
+    serverBindings: Optional[Dict[str, Union[ServerBinding, Reference]]]
+    channelBindings: Optional[Dict[str, Union[ChannelBinding, Reference]]]
+    operationBindings: Optional[Dict[str, Union[OperationBinding, Reference]]]
+    messageBindings: Optional[Dict[str, Union[MessageBinding, Reference]]]
+
+
+class AsyncAPI(BaseModel):
+    asyncapi: str
+    id: Optional[str]
+    info: Info
+    servers: Optional[Dict[str, Server]]
+    defaultContentType: Optional[str]
+    channels: Channels
+    components: Optional[Components]
+    tags: Optional[List[Tag]]
+    externalDocs: Optional[ExternalDocumentation]
```

Comment:
https://www.asyncapi.com/docs/specifications/v2.3.0#specificationExtensions

```suggestion
    externalDocs: Optional[ExternalDocumentation]
    
    class Config:
        extra = "allow"
```

Reply:
You can resolve when applied.